st2common/st2common/persistence/trigger.py

```diff
@@ -52,15 +52,24 @@ def delete_if_unreferenced(cls, model_object, publish=True, dispatch_trigger=Tru
         delete_query = model_object._object_key
         delete_query['ref_count__lte'] = 0
         cls._get_impl().delete_by_query(**delete_query)
+
+        # Since delete_by_query cannot tell if teh delete actually happened check with a get call
+        # if the trigger was deleted. Unfortuantely, this opens up to races on delete.
+        confirmed_delete = False
+        try:
+            cls.get_by_id(model_object.id)
+        except ValueError:
```

Comment:
This is the raised exception https://github.com/StackStorm/st2/blob/master/st2common/st2common/models/db/__init__.py#L126

Cannot handle anything else at this level. If I were to pick an exception to re-raise it would be StackstormDBObjectNotFound or something.